cinema/views.py

```diff
@@ -56,3 +81,48 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset.select_related("movie", "cinema_hall")
+
+        if self.action == "list":
+            queryset = queryset.prefetch_related("tickets").annotate(
+                tickets_available=(
+                    F("cinema_hall__rows")
+                    * F("cinema_hall__seats_in_row")
+                    - Count("tickets")
+                )
+            )
+
+        movie_id = self.request.query_params.get("movie")
```

Comment:
Should this filtering be for each action?

Reply:
yeah makes sense to make it only for list action